cli/lsp/logging.rs

```diff
@@ -53,7 +105,11 @@ macro_rules! lsp_log {
     if lsp_log_level == log::Level::Debug {
       $crate::lsp::logging::lsp_debug!($($arg)+)
     } else {
-      log::log!(lsp_log_level, $($arg)+)
+      let s = std::format!($($arg)+);
+      if let Some(log_file) = &mut *$crate::lsp::logging::log_file().lock() {
```

Comment:
Is there a way we could rewrite this to not acquire a mutex when this feature is disabled?